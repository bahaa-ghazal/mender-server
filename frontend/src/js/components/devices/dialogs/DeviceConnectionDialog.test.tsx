// Copyright 2019 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import { defaultState, render } from '@/testUtils';
import { undefineds } from '@northern.tech/testing/mockData';
import { act, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { vi } from 'vitest';

import DeviceConnectionDialog from './DeviceConnectionDialog';

describe('DeviceConnectionDialog Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<DeviceConnectionDialog onCancel={vi.fn} />);
    const view = baseElement.getElementsByClassName('MuiDialog-root')[0];
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as intended', async () => {
    const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime });
    render(<DeviceConnectionDialog onCancel={vi.fn} />, {
      preloadedState: {
        ...defaultState,
        app: {
          ...defaultState.app,
          features: {
            ...defaultState.app.features,
            isHosted: true
          }
        }
      }
    });
    await user.click(screen.getByText(/get started/i));
    expect(screen.getByText(/Enter your device type/i)).toBeInTheDocument();
    await user.click(screen.getByRole('button', { name: /back/i }));
    await user.click(screen.getByText(/Try a virtual device/i));
    expect(screen.getByText(/run the following command to start the virtual device/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Waiting for device/i })).toBeInTheDocument();
    await act(async () => vi.runAllTicks());
  });
});
