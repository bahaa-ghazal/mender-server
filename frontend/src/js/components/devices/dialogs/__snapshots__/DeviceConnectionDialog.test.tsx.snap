// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeviceConnectionDialog Component > renders correctly 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

@media print {
  .emotion-0 {
    position: absolute!important;
  }
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  height: 100%;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media print {
  .emotion-2 {
    height: auto;
  }
}

.emotion-3 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: var(--Paper-shadow);
  background-image: var(--Paper-overlay);
  margin: 32px;
  position: relative;
  overflow-y: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: calc(100% - 64px);
  max-width: 600px;
}

@media print {
  .emotion-3 {
    overflow-y: visible;
    box-shadow: none;
  }
}

@media (max-width:663.95px) {
  .emotion-3.MuiDialog-paperScrollBody {
    max-width: calc(100% - 64px);
  }
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  padding: 16px 24px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-overflow-scrolling: touch;
  overflow-y: auto;
  padding: 20px 24px;
}

.MuiDialogTitle-root+.emotion-5 {
  padding-top: 0;
}

.emotion-6 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-7 {
  background-color: #f5f5f5;
}

.emotion-7 .os-list img {
  height: 80px;
  margin: 16px;
}

.emotion-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border: 0;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: var(--variant-containedColor);
  background-color: var(--variant-containedBg);
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  --variant-textColor: #9c27b0;
  --variant-outlinedColor: #9c27b0;
  --variant-outlinedBorder: rgba(156, 39, 176, 0.5);
  --variant-containedColor: #fff;
  --variant-containedBg: #9c27b0;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-10::-moz-focus-inner {
  border-style: none;
}

.emotion-10.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-10 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-10:hover {
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-10:hover {
    box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  }
}

.emotion-10:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-10.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-10.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: hover) {
  .emotion-10:hover {
    --variant-containedBg: #7b1fa2;
    --variant-textBg: rgba(156, 39, 176, 0.04);
    --variant-outlinedBorder: #9c27b0;
    --variant-outlinedBg: rgba(156, 39, 176, 0.04);
  }
}

.emotion-10.MuiButton-loading {
  color: transparent;
}

.emotion-11 {
  border: 1px solid rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  gap: 8px;
}

.emotion-12 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
  margin-bottom: 0.35em;
}

.emotion-13 {
  height: 40px;
  margin-left: 16px;
}

.emotion-15 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-16 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border: 0;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 6px 8px;
  color: var(--variant-textColor);
  background-color: var(--variant-textBg);
  --variant-textColor: #1976d2;
  --variant-outlinedColor: #1976d2;
  --variant-outlinedBorder: rgba(25, 118, 210, 0.5);
  --variant-containedColor: #fff;
  --variant-containedBg: #1976d2;
  padding: 4px 5px;
  font-size: 0.8125rem;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-16::-moz-focus-inner {
  border-style: none;
}

.emotion-16.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-16 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-16:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

@media (hover: hover) {
  .emotion-16:hover {
    --variant-containedBg: #1565c0;
    --variant-textBg: rgba(25, 118, 210, 0.04);
    --variant-outlinedBorder: #1976d2;
    --variant-outlinedBg: rgba(25, 118, 210, 0.04);
  }
}

.emotion-16.MuiButton-loading {
  color: transparent;
}

.emotion-20 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
}

.emotion-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.emotion-24>:not(style)~:not(style) {
  margin-left: 8px;
}

.emotion-25 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  fill: currentColor;
  font-size: 1.5rem;
  color: #1976d2;
}

.emotion-25.read {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-26 {
  position: absolute;
  top: -5px;
  bottom: 0;
  left: -5px;
  right: -5px;
  border: 1px dashed #1976d2;
  border-radius: 50%;
}

.emotion-26.read {
  border-color: rgba(0, 0, 0, 0.38);
}

.emotion-27 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border: 0;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 6px 8px;
  color: var(--variant-textColor);
  background-color: var(--variant-textBg);
  --variant-textColor: #1976d2;
  --variant-outlinedColor: #1976d2;
  --variant-outlinedBorder: rgba(25, 118, 210, 0.5);
  --variant-containedColor: #fff;
  --variant-containedBg: #1976d2;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-27::-moz-focus-inner {
  border-style: none;
}

.emotion-27.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-27 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-27:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-27.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

@media (hover: hover) {
  .emotion-27:hover {
    --variant-containedBg: #1565c0;
    --variant-textBg: rgba(25, 118, 210, 0.04);
    --variant-outlinedBorder: #1976d2;
    --variant-outlinedBg: rgba(25, 118, 210, 0.04);
  }
}

.emotion-27.MuiButton-loading {
  color: transparent;
}

<div
  class="MuiDialog-root MuiModal-root emotion-0"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop emotion-1"
    style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper emotion-2"
    role="presentation"
    style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby="«r1»"
      aria-modal="true"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm emotion-3"
      role="dialog"
      style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root emotion-4"
        id="«r1»"
      >
        Connecting a device
      </h2>
      <div
        class="MuiDialogContent-root emotion-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-6"
        >
          You can connect almost any device and Linux OS with Mender, but to make things simple during evaluation we recommend you use a Raspberry Pi as a test device.
        </p>
        <div
          class="margin-top-small padding-small rpi-quickstart emotion-7"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 emotion-8"
          >
            Raspberry Pi quick start
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-6"
          >
            We'll walk you through the steps to connect a Raspberry Pi and deploy your first update with Mender.
          </p>
          <div
            class="flexbox column centered"
          >
            <div
              class="flexbox centered os-list"
            >
              <img
                src="/src/assets/img/raspberrypi.png"
              />
              <img
                src="/src/assets/img/raspberrypi4.png"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary emotion-10"
              tabindex="0"
              type="button"
            >
              Get started
            </button>
          </div>
        </div>
        <div
          class="two-columns margin-top-small"
        >
          <div
            class="padding-small padding-bottom-none flexbox column emotion-11"
          >
            <div
              class="flexbox center-aligned"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom emotion-12"
              >
                Use a virtual device
              </h6>
              <img
                class="emotion-13"
                src="/src/assets/img/docker.png"
              />
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 emotion-6"
            >
              Don't have a Raspberry Pi?
              <br />
              You can use our Docker-run virtual device to go through the same tutorial.
            </p>
            <div>
              <p
                class="MuiTypography-root MuiTypography-body1 emotion-15"
              >
                If you want to evaluate our commercial components such as mender-monitor, please use a physical device instead as the virtual client does not support these components at this time.
              </p>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary emotion-16"
                tabindex="0"
                type="button"
              >
                Try a virtual device
              </button>
            </div>
          </div>
          <div
            class="padding-small emotion-11"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom emotion-12"
            >
              Other devices
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 emotion-6"
            >
              See the documentation to integrate the following with Mender:
            </p>
            <ul
              class="MuiList-root MuiList-padding emotion-20"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters padding-top-none padding-bottom-none emotion-21"
              >
                <a
                  class=""
                  href="https://docs.mender.io/operating-system-updates-debian-family"
                  rel="noopener"
                  target="_blank"
                >
                  Debian family
                </a>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters padding-top-none padding-bottom-none emotion-21"
              >
                <a
                  class=""
                  href="https://docs.mender.io/operating-system-updates-yocto-project"
                  rel="noopener"
                  target="_blank"
                >
                  Yocto OSes
                </a>
              </li>
            </ul>
            <p
              class="MuiTypography-root MuiTypography-body1 emotion-6"
            >
              Or visit 
              <a
                href="https://hub.mender.io/c/board-integrations"
                rel="noopener"
                target="_blank"
              >
                Mender Hub
              </a>
               
              and search integrations for your device and OS.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing emotion-24"
      >
        <div
          class=""
          data-mui-internal-clone-element="true"
          style="margin-left: 20px;"
        >
          <div
            class="relative"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium emotion-25"
              data-testid="HelpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 17h-2v-2h2zm2.07-7.75-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25"
              />
            </svg>
            <div
              class="emotion-26 "
            />
          </div>
        </div>
        <div
          style="flex-grow: 1;"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary emotion-27"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
